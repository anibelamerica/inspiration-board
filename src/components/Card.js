import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    const { text, emoji } = this.props.card;
    const emojiLib = require("emoji-dictionary");

    return (
      <div className="card">
        <div className="card__content">
          { text &&
            <h1 className='card__content-text'>{text}</h1>}
          { emoji &&
            <div className='card__content-emoji'>{emojiLib.getUnicode(emoji)}</div>}
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  card: PropTypes.object,
  // text: PropTypes.string,
  // emoji: PropTypes.string,
};

export default Card;
