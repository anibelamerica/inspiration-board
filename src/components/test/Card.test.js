import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it renders even when passed an empty object prop', () => {
    const wrapper = shallow(
      <Card
        card={{}}/>);

    expect(wrapper).toMatchSnapshot();
  });

  test('that it renders when passed an object prop with text and emoji fields', () => {
    const wrapper = shallow(
      <Card
        card={
          {
            card: {
              text: "test inspirational text",
              emoji: "dog"
            }
          }
        }
        />);

    expect(wrapper).toMatchSnapshot();
  });
});
