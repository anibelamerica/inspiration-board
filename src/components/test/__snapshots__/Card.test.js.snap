// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it renders even when passed an empty object prop 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <div>
      <button
        className="card__delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`Card that it renders when passed an object prop with text and emoji fields 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <div>
      <button
        className="card__delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
